docs/fold.md

```diff
@@ -0,0 +1,84 @@
+## AIR: Fold instruction
+Here is the `fold` syntax.
+
+`(fold <iterable> <iterator> <instruction> [<last_instruction>])`
```

Comment:
How does `fold` handle errors in its `instruction`?

Reply:
Only fold over stream influences error, actually it supresses them, otherwise it'd be not deterministic. Fold over scalar and canonicalized maps and streams doesn't influence errors anyhow, the best analogy is the provided one with a tree, so it behaves exactly like in this tree representation.

Reply:
can we dump answers into documentation too?